platforms/android/tangram/src/main/java/com/mapzen/tangram/MapController.java

```diff
@@ -480,7 +480,11 @@ public void flyToCameraPosition(@NonNull final CameraPosition position, @Nullabl
 
     private void flyToCameraPosition(@NonNull final CameraPosition position, final int duration, @Nullable final CameraAnimationCallback callback, final float speed) {
         checkPointer(mapPointer);
-        // TODO: Make sense to have mark animating for flyTo irrespective of duration/speed?
+        if (duration == 0) {
+            setMapRegionState(MapRegionChangeState.JUMPING);
+        } else {
+            setMapRegionState(MapRegionChangeState.ANIMATING);
+        }
         setMapRegionState(MapRegionChangeState.ANIMATING);
```

Comment:
Probably want to remove this line since you added the branches above.

Reply:
Yeah will clean.